### PR TITLE
Add protobuf generation CI checks to avoid situations where our protobuf objects do not match our definitions.

### PR DESCRIPTION
**Pull request by surfnerd**
Add CI checks to ensure our protobufs are not stale.  And upload a patch artifact that can be used to apply changes locally.

### CODE REVIEW

File: UnitySDK/ProjectSettings/XRSettings.asset [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "m_SettingKeys": [
```
Comment (chriselion) [inline UnitySDK/ProjectSettings/XRSettings.asset]: Remove this too?

File: protobuf-definitions/make.sh [hunk 2]
```diff
@@ -21,8 +21,8 @@ mkdir -p $DST_DIR_P/$PYTHON_PACKAGE
 
 # generate proto objects in python and C#
 
-protoc --proto_path=proto --csharp_out=$DST_DIR_C $SRC_DIR/*.proto
-protoc --proto_path=proto --python_out=$DST_DIR_P --mypy_out=$DST_DIR_P $SRC_DIR/*.proto
+$COMPILER/protoc --proto_path=proto --csharp_out=$DST_DIR_C $SRC_DIR/*.proto
```
Comment (chriselion) [inline protobuf-definitions/make.sh]: Should we update the comments above? Anything in https://github.com/Unity-Technologies/ml-agents/blob/develop/protobuf-definitions/README.md that can be cleaned up?

File: .circleci/config.yml [hunk 3]
```diff
@@ -120,6 +120,53 @@ jobs:
...
+      - run:
+          name: Generate protobufs
+          command: |
+            . venv/bin/activate
+            cd protobuf-definitions
+            chmod +x Grpc.Tools.1.14.1/tools/linux_x64/protoc
+            chmod +x Grpc.Tools.1.14.1/tools/linux_x64/grpc_csharp_plugin
+            COMPILER=Grpc.Tools.1.14.1/tools/linux_x64 ./make.sh
+            git diff --exit-code --quiet -- :/ ':(exclude,top)UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects/*.meta' \
+              || { GIT_ERR=$?; echo "protobufs need to be regenerated, apply the patch updloaded to artifacts."; \
```
Comment (chriselion) [inline .circleci/config.yml]: nit: `updloaded` -> `uploaded`
Can you add the command necessary to apply the patch in the error message? Is it just `git apply proto.patch`?

Comment (chriselion) [reply .circleci/config.yml]: Is it worth moving some of this logic to the `.sh` file? Maybe enabled with a commandline arg/env variable?

File: protobuf-definitions/make.sh [hunk 4]
```diff
@@ -10,7 +10,7 @@ SRC_DIR=proto/mlagents/envs/communicator_objects
 DST_DIR_C=../UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects
 DST_DIR_P=../ml-agents-envs
 PROTO_PATH=proto
-
+CS_CODE_PATH=../UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects
```
Comment (chriselion) [inline protobuf-definitions/make.sh]: nit: same as `DST_DIR_C` above?

File: .circleci/config.yml (hunk 3 above)
Comment (surfnerd) [reply .circleci/config.yml]: yes, it is `git apply proto.patch`.  I'm not sure it makes sense to me to do git checks in the `.sh` that we will only need for CI.

File: protobuf-definitions/make.sh (hunk 4 above)
Comment (surfnerd) [reply protobuf-definitions/make.sh]: whoops

File: protobuf-definitions/make.sh (hunk 2 above)
Comment (surfnerd) [reply protobuf-definitions/make.sh]: Yes, now that the protobufs have been regenerated, I'm going to update the docs.

**Pull request merged**